File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var zeroFilledSubarray = function(nums) {
    let ans=0;
    let count=0;
    for(let i of nums){
        if(i===0){
            count++;
        }
        else{
            ans+=(count*(count+1)/2);
            count=0;
        }
    }
    ans+=(count*(count+1)/2);
    return ans;
};

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long ans=0;
        long long count=0;
        for(int i=0,n=nums.size();i<n;i++)
        {
            if(nums[i]==0)
            {
                count++;
            }
            else
            {
                ans+=(count*(count+1)/2);
                count=0;
            }
        }
        return ans+1ll*(count*(count+1)/2);
    }
};